### PR TITLE
feat(FTL-16871): integrate with redirect flow

### PR DESCRIPTION
**Pull request by rbrazhnyk**

### CODE REVIEW

File: samples/affinidi-nextjs-nextauthjs/src/lib/clients/iota.ts [hunk 1]
```diff
@@ -2,27 +2,97 @@ import {
...
+  CallbackApi,
+  IotaConfigurationDtoModeEnum,
+  InitiateDataSharingRequestOKData,
+  FetchIOTAVPResponseOK,
 } from "@affinidi-tdk/iota-client";
+import { v4 as uuidv4 } from "uuid";
+import * as jose from "jose";
 import { getAuthProvider } from "./auth-provider";
 
 export async function listIotaConfigurations() {
```
Comment (maindotdev) [inline samples/affinidi-nextjs-nextauthjs/src/lib/clients/iota.ts:15]: We could name this function `listIotaWebsocketConfigurations` to make it more meaningful in terms of the return value.

File: samples/auth0-nextjs-nextauthjs/src/lib/clients/iota.ts [hunk 2]
```diff
@@ -2,27 +2,97 @@ import {
...
+  CallbackApi,
+  IotaConfigurationDtoModeEnum,
+  InitiateDataSharingRequestOKData,
+  FetchIOTAVPResponseOK,
 } from "@affinidi-tdk/iota-client";
+import { v4 as uuidv4 } from "uuid";
+import * as jose from "jose";
 import { getAuthProvider } from "./auth-provider";
 
 export async function listIotaConfigurations() {
```
Comment (maindotdev) [inline samples/auth0-nextjs-nextauthjs/src/lib/clients/iota.ts:15]: Same as in previous comment.

**Review by maindotdev — DISMISSED** [review]

**Review by andrew-affinidi — APPROVED** [review]

**Review by maindotdev — APPROVED** [review]

File: generator/nextjs/template/src/components/iota/RedirectFlowPage.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    localStorage.setItem("iotaRedirect", JSON.stringify(toStore));
+
+    const vaultLink = getShareLink(data.jwt, "client_id");
+    router.push(vaultLink);
+  }
+
+  function getShareLink(jwt: string, clientId: string) {
+    return VaultUtils.buildShareLink(jwt, clientId);
+  }
```
Comment (carlos-affinidi) [inline generator/nextjs/template/src/components/iota/RedirectFlowPage.tsx:98]: No need for a separate function

File: generator/nextjs/template/src/pages/iota-callback.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import dynamic from "next/dynamic";
+import { personalAccessTokenConfigured } from "src/lib/env";
+
+const IotaRedirectFlowCallbackPage = dynamic(
+  () => import("../components/iota/RedirectFlowCallbackPage"),
+  { ssr: false },
```
Comment (carlos-affinidi) [inline generator/nextjs/template/src/pages/iota-callback.tsx:7]: There should be no need to make the redirect pages components and disable SSR.

File: generator/nextjs/template/src/pages/iota-redirect.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import dynamic from "next/dynamic";
+import { personalAccessTokenConfigured } from "src/lib/env";
+
+const IotaRedirectFlowPage = dynamic(
+  () => import("../components/iota/RedirectFlowPage"),
+  { ssr: false },
```
Comment (carlos-affinidi) [inline generator/nextjs/template/src/pages/iota-redirect.tsx:7]: There should be no need to make the redirect pages components and disable SSR.

**Pull request merged**